appinventor/appengine/war/index.jsp

```diff
@@ -1,3 +1,20 @@
+<%@page import="com.google.appinventor.server.Server,com.google.appinventor.common.version.AppInventorFeatures" %>
+<%
+   if (request.getScheme().equals("http") && Server.isProductionServer()
+       && AppInventorFeatures.enableHttpRedirect()) {
+        String qs = request.getQueryString();
+        String host = request.getServerName();
```

Comment:
The problem is which port to use. We *cannot* use the port that the connection just came in on because it isn't using https (if it was, we wouldn't get here). The best we could do is have a configuration option to specify the port to use for https if not the standard one. I don't believe it is worth doing at this time.

Reply:
That's a fair point.